go/test/endtoend/utils/utils.go

```diff
@@ -236,16 +236,17 @@ func WaitForAuthoritative(t *testing.T, ks, tbl string, readVSchema func() (*int
 		case <-timeout:
 			return fmt.Errorf("schema tracking didn't mark table t2 as authoritative until timeout")
 		default:
-			time.Sleep(1 * time.Second)
```

Comment:
It's better to wait when we know we need to actually retry, so the tests are much faster in their happy path. Found this since I was running the racy test with a high number of counts and this speeds that up hugely. Now it takes 18 seconds for the full test to run 100 times, vs 120 seconds before. 

I have applied the same optimization to the other helpers here and reduced the retry time to 100ms instead of 1s to also speed it up in the failure path.